src/main/src/main/java/org/geoserver/catalog/CoverageViewPamResourceInfo.java

```diff
@@ -0,0 +1,98 @@
+/* (c) 2024 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.catalog;
+
+import it.geosolutions.imageio.pam.PAMDataset;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.geotools.api.coverage.grid.GridCoverageReader;
+import org.geotools.api.data.ResourceInfo;
+import org.geotools.coverage.grid.io.GridCoverage2DReader;
+import org.geotools.coverage.grid.io.PAMResourceInfo;
+import org.geotools.data.DefaultResourceInfo;
+
+/** A ResourceInfo wrapper for CoverageViews that provides access to the PAMDataset */
+public class CoverageViewPamResourceInfo extends DefaultResourceInfo implements PAMResourceInfo {
+    private final PAMDataset viewPam;
+
+    /**
+     * Constructor that populates the PAM dataset from the view bands
+     *
+     * @param inputAlphaNonNullCoverages input alpha non-null coverages
+     */
+    public CoverageViewPamResourceInfo(
```

Comment:
Does not need like you need the coverages to build the "info" object, so I'd avoid collecting them, reading the raster can allocate actual memory for nothing, depending on the data source (e.g., GeoTIFF does delayed loading, but NetCDF does not).

Reply:
coverages instantiation made optional